Add fvcom simulator (#1572)
inductiva/inductiva: docs/simulators/FVCOM.md

# FVCOM

[FVCOM](https://github.com/CaNS-World/CaNS) (Finite Volume Community Ocean Model)
is a 3D hydrodynamic model designed for simulating coastal and ocean dynamics.
It utilizes an unstructured grid with finite-volume methods, providing flexibility
in modeling complex coastlines, estuaries, and varying bathymetry. FVCOM excels
in simulating ocean circulation, tides, and coastal processes, offering
high-resolution outputs for water currents, temperature, salinity, and ecosystem
interactions. Its versatility and precision make it an essential tool for
studying coastal environments, marine ecosystems, and the impacts of climate
change on ocean systems.

We have compiled two versions of FVCOM: the standard **fvcom** binary and an

> additional **fvcom_ESTUARY** binary, which is configured to run the Estuary test

Reviewer: ```suggestion
additional **fvcom_estuary** binary, which is configured to run the Estuary test
```

> case included in the FVCOM package. Users can check the compilation flags used
> for each binary in the respective `make.inc` files located at `/make.inc` and

Reviewer: ```suggestion
case included in the FVCOM package. Users can check the compilation modules used
in each binary in the respective `make.inc` files located at `/make.inc` and
```

> `/make_estuary.inc`.

Reviewer: maybe share where these files can be found?

> `/make_ESTUARY.inc`.

Reviewer: ```suggestion
`/make_estuary.inc`.
```

These are the only FVCOM versions available at the moment, but we are working on
a solution to allow users to customize and specify their own flags for on-the-fly
compilation.

> If you are having trouble with the nml file you can do the following:

Reviewer: ```suggestion
If you are having trouble with the input namelist file, you can do the following to
generate a valid `nml` file in the `working_dir` with the expected format for
the requested model:
```

```python
import inductiva

# Set simulation input directory
input_dir = inductiva.utils.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "fvcom-input-example.zip", unzip=True)

# Initialize the Simulator
fvcom = inductiva.simulators.FVCOM()

# Run simulation with config files in the input directory
task = fvcom.run( input_dir=input_dir,
                  working_dir="run/",
                  create_namelist="tst",
                  n_vcpus=1)

task.wait()
task.download_outputs()

```
This will generate a nml file in the `working_dir` with the expected format for

> that binary.

Reviewer: You can delete this line.

> ## Example

Reviewer: add a description of what this example is doing (running the default model with debug level 7 and. 1 MPI porcess, ...)
